doc/python_api_reference_vDev.md

```diff
@@ -2437,6 +2438,77 @@ def search_for_undetectable_logical_errors(
     """
 ```
 
+<a name="stim.Circuit.shortest_error_problem_as_wcnf_file"></a>
+```python
+# stim.Circuit.shortest_error_problem_as_wcnf_file
+
+# (in class stim.Circuit)
+def shortest_error_problem_as_wcnf_file(
+    self,
+    *,
+    weighted: bool = False,
+    weight_scale_factor: int = 0,
+) -> str:
+    """Generates a maxSAT problem instance in WDIMACS format whose optimal value is
```

Comment:
Stim's docstrings follow the convention that the first line is a self-contained paragraph summarizing the method.

I've made an attempt at improving the docstring:

```
Makes a maxSAT problem of the circuit's distance, that other tools can solve.

The output is a string describing the maxSAT problem in WDIMACS format
(see https://maxhs.org/docs/wdimacs.html). The optimal solution to the
problem is the fault distance of the circuit (the minimum number of error
mechanisms that combine to flip any logical observable while producing no
detection events).

There are many tools that can solve maxSAT problems in WDIMACS format.
For example, you can download one of the entries in the 2023 maxSAT
competition (see https://maxsat-evaluations.github.io/2023) and run it on
your problem by running these BASH terminal commands:

    wget https://maxsat-evaluations.github.io/2023/mse23-solver-src/exact/CASHWMaxSAT-CorePlus.zip
    unzip CASHWMaxSAT-CorePlus.zip
    ./CASHWMaxSAT-CorePlus/bin/cashwmaxsatcoreplus -bm -m your_problem.wcnf

Args:
    weight_bins: Defaults to 1 (unweighted errors). Error probabilities are converted to
        log-odds and scaled/rounded to be positive integers at most this large. Setting
        this argument to a larger number results in the solution to the problem being the
        most likely logical error weighted by probability of its parts, instead of just the
        one with the fewest parts.

Returns:
    A string corresponding to the contents of a WCNF file in WDIMACS format.

Examples:
    ...
```

Reply:
Thanks. I had actually omitted the url because I got this error:
```
E   ImportError: Docstring line has length 98 > 80:
E       wget https://maxsat-evaluations.github.io/2023/mse23-solver-src/exact/CASHWMaxSAT-CorePlus.zip
E   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Any idea how to get past it?

Reply:
Go to https://github.com/quantumlib/Stim/blob/1cdf45f884442d01806d6a8997b867a352e937c9/src/stim/cmd/command_help.cc#L74 and add `"wget "` as a prefix string that disables the check. Or change it to allow if `https://` appears anywhere instead of just at the start.

Reply:
Thank you!!

Reply:
That doesn't work. I still get:
```
E   ImportError: Docstring line has length 98 > 80:
E       wget https://maxsat-evaluations.github.io/2023/mse23-solver-src/exact/CASHWMaxSAT-CorePlus.zip
E   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
I think the error is happening later on when we try to import the module, not when we clean the docstring

Reply:
Ah nvm, I got it 🤦